test/e2e/config/config.go

```diff
@@ -39,6 +39,7 @@ type Config struct {
 	UseDeployCommand    bool   `envconfig:"USE_DEPLOY_COMMAND"`
 	GinkgoFocus         string `envconfig:"GINKGO_FOCUS"`
 	GinkgoSkip          string `envconfig:"GINKGO_SKIP"`
+	NewCluster          bool   `envconfig:"NEW_CLUSTER" default:"false"` // track when we create a new cluster via E2E
```

Comment:
isn't this equivalent to `Name == ""`?

Reply:
nevermind, this is needed because you are using it after Name is set. I don't think we should make it an envconfig var though since its value will be set in https://github.com/Azure/aks-engine/pull/33/files#diff-881ae33659acba7c8c55e6efbf5edd3dR107 with `os.Setenv("NEW_CLUSTER", "true")`. This isn't a configurable input but a variable determined by other values.

Reply:
I'm pretty sure that's the only way that the env config parser works. It has a whitelist of env vars to parse and turn into a typed struct.

Reply:
it can follow the same patter as `CurrentWorkingDir`